### PR TITLE
Add libaom AV1 codec.

### PR DESCRIPTION
**Pull request by contradict**

### CODE REVIEW

File: L/libaom/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+products = [
+    ExecutableProduct("aomenc", :aomenc),
+    ExecutableProduct("aomdec", :aomdec),
+    LibraryProduct(["libaom", "aom"], :libaom),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+                BuildDependency("YASM_jll"; platforms=filter(p->occursin("86", triplet(p)) && !occursin("apple", triplet(p)), platforms)),
+                BuildDependency("NASM_jll"; platforms=filter(p->occursin("86", triplet(p)) && occursin("apple", triplet(p)), platforms)),
```
Comment (giordano) [inline L/libaom/build_tarballs.jl:43]: I don't think these can be used as build dependencies like this: https://docs.binarybuilder.org/stable/build_tips/#Dependencies-for-the-target-system-vs-host-system

Comment (contradict) [reply L/libaom/build_tarballs.jl:43]: Oh, I see. I'll switch them to HostBuildDependencies.

File: L/libaom/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    LibraryProduct(["libaom", "aom"], :libaom),
+]
+
+# Dependencies that must be installed before this package can be built
+#
+# YASM is recommended in the build instructions, but errors on apple platforms.
+# Assembly only exists for x86 targets.
+dependencies = [
+                HostBuildDependency("YASM_jll"; platforms=filter(p->occursin("86", triplet(p)) && !occursin("apple", triplet(p)), platforms)),
+                HostBuildDependency("NASM_jll"; platforms=filter(p->occursin("86", triplet(p)) && occursin("apple", triplet(p)), platforms)),
```
Comment (giordano) [inline L/libaom/build_tarballs.jl:46]: This is a bit easier to read:
```suggestion
    HostBuildDependency("YASM_jll"; platforms=filter(p->proc_family(p) == "intel" && !Sys.isapple(p), platforms)),
    HostBuildDependency("NASM_jll"; platforms=filter(p->proc_family(p) == "intel" && Sys.isapple(p), platforms)),
```

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**